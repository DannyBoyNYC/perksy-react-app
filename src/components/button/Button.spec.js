const React = require('react');

const Button = require('./index').default;

describe('<Button.* />', () => {
  let component;
  const requiredProps = {
    children: 'Click me!'
  };

  describe('<Button.Primary />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Button.Primary {...requiredProps} />,
        'Button'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('add the disabled attribute when isDisabled is true', () => {
      component = shallowUntilTarget(
        <Button.Primary {...requiredProps} isDisabled />,
        'Button'
      );
      expect(component).toBeDisabled();
    });

    it('calls the given onClick method when clicked', () => {
      const mockedMethod = jest.fn();
      component = shallowUntilTarget(
        <Button.Primary {...requiredProps} onClick={mockedMethod} />,
        'Button'
      );
      component.simulate('click');
      expect(mockedMethod).toHaveBeenCalled();
    });
  });

  describe('<Button.Info />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Button.Info {...requiredProps} />,
        'Button'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Button.Success />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Button.Success {...requiredProps} />,
        'Button'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Button.Warning />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Button.Warning {...requiredProps} />,
        'Button'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('<Button.Danger />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <Button.Danger {...requiredProps} />,
        'Button'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
