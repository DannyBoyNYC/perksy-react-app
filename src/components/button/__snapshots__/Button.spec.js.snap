// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button.* /> <Button.Danger /> should render correctly with all required props 1`] = `
<button
  className="Button-root-0-1-1 Button-danger-0-1-6"
  type="button"
>
  Click me!
</button>
`;

exports[`<Button.* /> <Button.Info /> should render correctly with all required props 1`] = `
<button
  className="Button-root-0-1-1 Button-info-0-1-3"
  type="button"
>
  Click me!
</button>
`;

exports[`<Button.* /> <Button.Primary /> should render correctly with all required props 1`] = `
<button
  className="Button-root-0-1-1 Button-primary-0-1-2"
  type="button"
>
  Click me!
</button>
`;

exports[`<Button.* /> <Button.Success /> should render correctly with all required props 1`] = `
<button
  className="Button-root-0-1-1 Button-success-0-1-4"
  type="button"
>
  Click me!
</button>
`;

exports[`<Button.* /> <Button.Warning /> should render correctly with all required props 1`] = `
<button
  className="Button-root-0-1-1 Button-warning-0-1-5"
  type="button"
>
  Click me!
</button>
`;
